### PR TITLE
Fix locks

### PR DESCRIPTION
**Pull request by Half-Shot**

### CODE REVIEW

File: src/bot.ts [hunk 1]
```diff
@@ -142,17 +142,22 @@ export class DiscordBot {
         if (this.channelLocks[channel.id]) {
             return;
         }
-        let i: NodeJS.Timeout;
+
         const p = new Promise((resolve) => {
-            i = setInterval(resolve, this.config.limits.discordSendDelay);
-            this.channelLocks[channel.id] = {i, p};
+            if (!this.channelLocks[channel.id]) {
+                return;
```
Comment (Sorunome) [inline src/bot.ts]: resolve(); before return, just in case?

File: src/bot.ts [hunk 2]
```diff
@@ -142,17 +142,22 @@ export class DiscordBot {
...
+            const i = setInterval(resolve, this.config.limits.discordSendDelay);
+            this.channelLocks[channel.id].i = i;
         });
+
+        this.channelLocks[channel.id] = {i: null, p};
     }
 
     public unlockChannel(channel: Discord.Channel) {
         const lock = this.channelLocks[channel.id];
         if (lock) {
```
Comment (Sorunome) [inline src/bot.ts]: `if (lock && lock.i)`, that way the `!` in `lock.i!` is unneeded in next line

File: src/bot.ts (hunk 1 above)
Comment (Half-Shot) [reply src/bot.ts]: The return is designed to save us setting up a timer if we have already unlocked the channel before the promise is processed. It's astronomically unlikely, but we don't need to setup a timer if it's been deleted.

Comment (Sorunome) [reply src/bot.ts]: soru meany
```js
if (!this.channelLocks[channel.id]) {
    resolve();
    return;
}
```
that the promise still resolves if we don't need a timer

Comment (Half-Shot) [reply src/bot.ts]: ooooh, yes

**Pull request merged**